### PR TITLE
Add some example transforms (dNBR for now)

### PR DESCRIPTION
**Pull request by alexgleith**
* dNBR for now
* Probably should make a less controversial NDVI also

### CODE REVIEW

File: datacube_alchemist/_utils.py [hunk 1]
```diff
@@ -183,7 +183,7 @@ def _guess_region_code(ds: Dataset) -> str:
     # The region code is 50JPP.
     tile_match = RE_TILE_REGION_CODE.match(ds.metadata_doc["tile_id"])
     if not tile_match:
-        raise ValueError("No region code for dataset {}".format(d.id))
+        raise ValueError("No region code for dataset {}".format(ds.id))
```
Comment (omad) [inline datacube_alchemist/_utils.py:186]: We should start using custom exceptions more.

File: datacube_alchemist/transforms.py [hunk 2]
```diff
@@ -0,0 +1,45 @@
+from typing import Dict
+
+import numpy
+from datacube import Datacube
+from datacube.virtual import Measurement, Transformation
+from xarray import Dataset, merge
+
+
+class DeltaNBR(Transformation):
+    """Return NBR"""
```
Comment (omad) [inline datacube_alchemist/transforms.py:10]: This should describe what it does, what can be configured, what it works on and what it doesn't...

File: datacube_alchemist/transforms.py [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+    def compute(self, data) -> Dataset:
+        base_year = data.time.dt.year.values[0] - 1
+        if base_year == 2021:
+            base_year = 2020
+        if base_year == 2012:
+            base_year = 2013
+
+        dc = Datacube()
+        gm_data = dc.load(
+            product="ls8_nbart_geomedian_annual",
```
Comment (omad) [inline datacube_alchemist/transforms.py:29]: Maybe better to pass in *comparison product name* from the configuration? But, then I'm not sure about how the custom logic for selecting years fits...

**Review by omad — APPROVED** [review]

**Review by omad — APPROVED** [review]

File: datacube_alchemist/settings.py [hunk 4]
```diff
@@ -34,12 +33,11 @@ def _convert_write_data_settings(settings):
...
 class OutputSettings:
     location: str
-    dtype: np.dtype
-    nodata: int  # type depends on dtype
     write_data_settings: Optional[Mapping[str, str]] = attr.ib(
         converter=_convert_write_data_settings
     )
-    preview_image: Optional[List[str]] = None
+    nodata: Optional[int] = None
+    preview_image: Optional[Mapping[Any, Any]] = None
```
Comment (alexgleith) [inline datacube_alchemist/settings.py:40]: Hey @omad what do you think of this change?

It increases flexibility, but breaks existing configs... I can't decide if it is a good idea.

Comment (omad) [reply datacube_alchemist/settings.py:40]: I think it's a good idea. I'm not sure what options eo-datasets3 support for preview images, but we would want to expose all of them.

We could make this backwards compatible, but, I don't think it's worth the extra code.

**Comment by alexgleith** [comment]
This has been replaced with #89

**Pull request closed without merging**